Pod/Classes/PagingMenuController.swift

```diff
@@ -9,484 +9,570 @@
+		if case .Infinite = options.menuDisplayMode {
+			guard index == currentPage || index == previousIndex || index == nextIndex else { return false }
+		} else {
+			guard index >= previousIndex && index <= nextIndex else { return false }
+		}
+		return true
+	}
+	
+	// MARK: - Page calculator
+	
+	private var currentPagingViewPosition: PagingViewPosition {
+		guard let contentScrollView = contentScrollView else { return .Left }
+		let pageWidth = contentScrollView.frame.width
+		let order = Int(ceil((contentScrollView.contentOffset.x - pageWidth / 2) / pageWidth))
+		
+		if case .Infinite = options.menuDisplayMode {
+			return PagingViewPosition(order: order) ?? .Left
+		}
+		
+		// consider left edge menu as center position
+		guard currentPage == 0 && contentScrollView.contentSize.width < (pageWidth * CGFloat(visiblePagingViewNumber)) else {
+			return PagingViewPosition(order: order) ?? .Left
+		}
+		
+		return PagingViewPosition(order: order + 1) ?? .Left
+	}
+	
+	private func targetPage(tappedPage tappedPage: Int) -> Int {
+		guard case let .Standard(_, _, scrollingMode) = options.menuDisplayMode else { return tappedPage }
+		guard case .PagingEnabled = scrollingMode else { return tappedPage }
+		return tappedPage < currentPage ? currentPage - 1 : currentPage + 1
+	}
+	
+	// MARK: - Validator
+	
+	private func validateDefaultPage() {
+		guard options.defaultPage >= options.menuItemCount || options.defaultPage < 0 else { return }
+		
+		NSException(name: ExceptionName, reason: "default page is invalid", userInfo: nil).raise()
+	}
+	
+	private func validatePageNumbers() {
+		guard case .Infinite = options.menuDisplayMode else { return }
+		guard options.menuItemCount < visiblePagingViewNumber else { return }
+		
+		NSException(name: ExceptionName, reason: "the number of view controllers should be more than three with Infinite display mode", userInfo: nil).raise()
+	}
+	
+	// MARK: - Private
+	
+	private var lastVisiblePage: Int = 0
+	
+	/// The viewController that will be visible when scrolling ends
+	private var toViewController: UIViewController?
+	
+	private func viewCopyImageView(isFront isFront: Bool) -> UIImageView {
+		let imageView = UIImageView()
+		imageView.backgroundColor = .greenColor()
+		contentScrollView.addSubview(imageView)
+		imageView.translatesAutoresizingMaskIntoConstraints = false
+		
+		let toItem = isFront ? contentScrollView : pagingViewControllers!.last!.view
+		
+		for attribute in [NSLayoutAttribute.Top, .Leading, .Height, .Width] {
+			let toAttribute: NSLayoutAttribute
+			
+			if !isFront && attribute == .Leading {
+				toAttribute = .Trailing
+			} else {
+				toAttribute = attribute
+			}
+			
+			NSLayoutConstraint(item: imageView, attribute: attribute, relatedBy: .Equal, toItem: toItem, attribute: toAttribute, multiplier: 1, constant: 0).active = true
+		}
+		
+		return imageView
+	}
+	private lazy var frontImageView: UIImageView = self.viewCopyImageView(isFront: true)
```

Comment:
It's used to generate a uimageview of the old vc to smooth the transition to the new one (cos in the meantime if the controller refresh the data on viewWillAppear the appearance of the updated view controller can be different)